### PR TITLE
modify rendering appearance of "deprecated" and "versionchanged"

### LINKED ISSUE #156: Feature: support ..deprecated Sphinx directive
Same as https://github.com/pytorch/pytorch/issues/73007

The `.. deprecated ::` sphinx directive isn't visible enough in the rendered docs:

```
.. deprecated:: 0.12

    This is deprecated, use blah instead
```

will simply render as:

image

Could we make it more obvious, similar to `.. note::` or `.. warning::` ? These are much more visible:

image

### PR DESCRIPTION
**Pull request by ain-soph**
Fixes #156

### CODE REVIEW

**Review by NicolasHug — APPROVED** [review]
For ref from https://github.com/pytorch/pytorch_sphinx_theme/issues/156#issuecomment-1183589446, this currently renders as:

image


Thanks a lot @ain-soph , LGTM! I'd suggest to merge this now and start using `.. deprecated::` in our docs. Since we just released a new version, we'll have plenty of time to catch issues.

(As a side note, it'd be great to have somehting similar for `.. versionchanged`)

@svekars is this good to go on your side?

File: pytorch_sphinx_theme/static/css/theme.css
Comment (svekars) [inline pytorch_sphinx_theme/static/css/theme.css]: To be consistent with other admonitions, let's make `border-radius: 0` and `border: none`
Also, background color can be more like in the Warning or Deprecated notice in [the issue description](https://github.com/pytorch/pytorch_sphinx_theme/issues/156) or can try #fdede9 as we use it in other parts of the theme.

**Comment by svekars** [comment]
I like this! A few suggestions around the CSS - let's keep it consistent with the rest of the theme. Let's make the border radius: 0 and and border: none! @NicolasHug can you be appointed to test this and report any issues? =) Also, is this going to be a new wheel and what's the upgrade path for all libraries?

**Comment by NicolasHug** [comment]
> @NicolasHug can you be appointed to test this and reporting any issues? 

Sure, I'll start using it in torchvision ASAP

> Also, is this going to be a new wheel and what's the upgrade path for all libraries?

All libraries rely on the latest version of the theme, as it is on the GitHub `master` branch:

https://github.com/pytorch/vision/blob/418d8a6f106bad5ae08c14951105735ba67eec1e/docs/requirements.txt#L7

So as soon as this PR is merged it will propagate to all new doc builds (in PRs, in nightlies, etc.).

File: pytorch_sphinx_theme/static/css/theme.css
Comment (ain-soph) [reply pytorch_sphinx_theme/static/css/theme.css]: Just to note that currently it’s not an admonition but still keeps a changeset directive (same as original sphinx)

**Comment by ain-soph** [comment]
Current Looking:  
image

**Review by svekars — APPROVED** [review]
Love it!

**Comment by ain-soph** [comment]
Anyone with write access please merge this.

Appreciate that.

cc @malfet @msaroufim @brianjo

**Pull request merged**
